src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animalBited) {
+    if (animalBited instanceof Herbivore && animalBited.hidden === false) {
```

Comment:
The check for 'animalBited.hidden === false' is correct, but consider checking if 'animalBited' is not an instance of Carnivore as well, as per the task requirements. The method should not work if it is another carnivore.